lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.c

```diff
@@ -575,20 +575,44 @@ static void insert_sgw_c_teid_to_directoryd(
     sgw_s11_teid_t* s11_teid_p = NULL;
     LIST_FOREACH(s11_teid_p, &ue_context_p->sgw_s11_teid_list, entries) {
       if (s11_teid_p) {
-        if (s11_teid_p->entries.le_next) {
-          snprintf(
-              teidString + strlen(teidString),
-              sizeof(teidString) - strlen(teidString), "%u,",
-              s11_teid_p->sgw_s11_teid);
-        } else {
-          snprintf(
-              teidString + strlen(teidString),
-              sizeof(teidString) - strlen(teidString), "%u",
-              s11_teid_p->sgw_s11_teid);
-        }
+        if (teid_type == CP_TEID) {
+          teidlist[teid_list_idx++] = s11_teid_p->sgw_s11_teid;
+        } else if (teid_type == UP_TEID) {
+          uint8_t idx = 0;
+          sgw_eps_bearer_context_information_t* sgw_session_ctxt_p =
+              sgw_get_sgw_eps_bearer_context(s11_teid_p->sgw_s11_teid);
+          if (sgw_session_ctxt_p) {
+            for (; idx < BEARERS_PER_UE; idx++) {
+              if (sgw_session_ctxt_p->pdn_connection
+                      .sgw_eps_bearers_array[idx]) {
+                teidlist[teid_list_idx++] = sgw_session_ctxt_p->pdn_connection
+                                                .sgw_eps_bearers_array[idx]
+                                                ->s_gw_teid_S5_S8_up;
+              }
+            }  // end of bearer list
+          }    // check for sgw_session_ctxt_p
+        }      // switch based on teid type
       }
+    }  // end of s11_teid_list
+  }    // end of ue context
+
+  for (uint8_t idx = 0; idx < teid_list_idx; idx++) {
+    if (idx == teid_list_idx - 1) {
+      snprintf(
+          teidString + strlen(teidString),
+          sizeof(teidString) - strlen(teidString), "%u", teidlist[idx]);
+    } else {
+      snprintf(
+          teidString + strlen(teidString),
+          sizeof(teidString) - strlen(teidString), "%u,", teidlist[idx]);
     }
+  }
+  if (teid_type == CP_TEID) {
     directoryd_update_record_field(imsi_str, "sgw_c_teid", teidString);
+  } else if (teid_type == UP_TEID) {
+    directoryd_update_record_field(imsi_str, "sgw_u_teid", teidString);
```

Comment:
here you can use the same strategy
```
tedi_type_label = sgw_c_teid;
if (teid_type == UP_TEID){
	tedi_type_label = sgw_u_teid;
}
directoryd_update_record_field(imsi_str, tedi_type_label, teidString);
```

Reply:
In c, we can not assign the string directly instead we need to call string function, strcpy(). Instead of calling  strcpy(), we can use the existing logic.

Reply:
> In c, we can not assign the string directly instead we need to call string function, strcpy(). Instead of calling strcpy(), we can use the existing logic.

Ignore above statement, it's doable without strcpy()